modules/swagger-codegen/src/main/resources/cpprest/api-source.mustache

```diff
@@ -53,7 +53,22 @@ pplx::task<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/r
     utility::string_t responseHttpContentType;
     
     // use JSON if possible
-    if ( responseHttpContentTypes.size() == 0 || responseHttpContentTypes.find(U("application/json")) != responseHttpContentTypes.end() )
+    if ( responseHttpContentTypes.size() == 0 )
+    {
+        {{#vendorExtensions.x-codegen-response.isPrimitiveType}}
+        responseHttpContentType = U("application/json");
+        {{/vendorExtensions.x-codegen-response.isPrimitiveType}}
+        {{^vendorExtensions.x-codegen-response.isPrimitiveType}}
+        {{#vendorExtensions.x-codegen-response.isString}}
```

Comment:
Adding isString looks considerably more complex, but doable. The question that comes to mind is if this has not been needed until now, is it the right feature to add the the CodegenResponse for a single language generator?